### PR TITLE
Comestibles support inheritance

### LINKED ISSUE #16094: Crafting Cooking oil causes the program to become non responsive.
0.C-12726-gcfc59e9
Win x64

Save directory:
https://mega.nz/#!qwFlgYoJ!QkVdKwRwRlDdBYY6PX5j9BE-bJzfGwvfW-7MBtORSRw

When I craft cooking oil, it does not show any containers I have in my inventory or nearby.  This results in me having to dump the oil on the floor, which crashed the game.  I don't get any debug messages.

It does, however show the container if I hold it.  If I put it into a held container, ie a gallon jug, I get a "Gallon jug of cooking oil" with no listed amount. 

Wishing for cooking oil results in  "cooking oil (0)", or "Gallon jug of cooking oil" if I wish for it in a container.

Neither shows up in my crafting menu as usable.

I have tried making a new world with new survivor, same results.

### PR DESCRIPTION
**Pull request by mugling**
Precursor to starting to implement vitamins/nutritional deficiencies

Incidentally fixes #16094

### CODE REVIEW

File: src/itype.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ struct islot_comestible
     std::string tool = "null";
 
     /** Defaults # of charges (drugs, loaf of bread? etc) */
-    long def_charges = 0;
+    long def_charges = 1;
```
Comment (Coolthulhu) [inline src/itype.h]: Does it work correctly with un-charged comestibles?

Comment (mugling) [inline src/itype.h]: These shouldn't exist and a quick check with `jq` confirms there are no items dependent upon a default value of `charges 0`

**Comment by Coolthulhu** [comment]
An example of inherited comestible would be useful. For example, orange juice inheriting from fruit juice.

**Comment by mugling** [comment]
> An example of inherited comestible would be useful. For example, orange juice inheriting from fruit juice.

Irradiated food is a good example but this is best changed en-masse. I want however to avoid extending this PR as I've a number of other branches I want to push are dependent upon it and it will cut the merge conflicts.

**Pull request merged**